### PR TITLE
Seafile-client update, updated python compatibility, libressl support

### PR DESCRIPTION
**Pull request by CaseOf**
Hello,
Here is a version bumped ebuild for net-misc/seafile and net-misc/seafile-client for version 7.0.9.
I have tested net-misc/seafile and net-libs/libsearpc with python compatibility for python 3.7 and it works then I updated PYTHON_COMPAT on these ebuilds.
net-misc/seafile builds fine with dev-libs/libressl and net-misc/seafile-client too with patched condition for version testing.

### CODE REVIEW

**Comment by evdenis** [comment]
Hi, maybe you will find some useful info in this PR https://github.com/gentoo/gentoo/pull/15884

**Comment by evdenis** [comment]
If you would like to maintain net-misc/seafile, net-misc/seafile-client and other maintainers have no objections you can take the maintainership of these packets. Currently, I'm not using gentoo and it's hard for me to update these packages.

**Comment by CaseOf** [comment]
I’ve tested only with Qt 5.14 to be honest.
Also, it’s not the first package I’ve heard to have issues with Qt 5.15.

**Comment by CaseOf** [comment]
About Qt 5.15 issues, I guess the best thing to do is to report upstream.

**Comment by CaseOf** [comment]
Opened an issue here for info: https://github.com/haiwen/seafile-client/issues/1310

File: net-libs/libsearpc/libsearpc-3.2.0-r1.ebuild [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 EAPI=7
 
-PYTHON_COMPAT=(python3_6)
+PYTHON_COMPAT=(python3_{6,7})
```
[the remark this answers is not in the document]
Comment (thesamesam) [reply net-libs/libsearpc/libsearpc-3.2.0-r1.ebuild:6]: ping.

File: net-misc/seafile-client/files/seafile-client-7.0.9-libressl.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/src/utils/rsa.cpp b/src/utils/rsa.cpp
```
[the remark this answers is not in the document]
Comment (thesamesam) [reply net-misc/seafile-client/files/seafile-client-7.0.9-libressl.patch:2]: ping

[the remark this answers is not in the document]
Comment (CaseOf) [reply net-misc/seafile-client/files/seafile-client-7.0.9-libressl.patch:2]: Github did not set this as obsolete but I added link before that line, you can check

File: net-libs/libsearpc/libsearpc-3.2.0-r1.ebuild (hunk 1 above)
[the remark this answers is not in the document]
Comment (CaseOf) [reply net-libs/libsearpc/libsearpc-3.2.0-r1.ebuild:6]: I admit that I did not take time to test it under py3.8 yet

**Comment by CaseOf** [comment]
finally added support for py3.8 and py3.9 after checking that works!

**Comment by CaseOf** [comment]
added myself as maintainer in these packages

**Comment by CaseOf** [comment]
fixed merge issue with master

**Comment by CaseOf** [comment]
update from upstream about Qt 5.15
https://github.com/haiwen/seafile-client/issues/1310#issuecomment-698699306

**Review by juippis — COMMENTED** [review]
```
/var/tmp/portage/net-misc/seafile-client-7.0.9/work/seafile-client-7.0.9/src/ui/private-share-dialog.cpp: In member function 'virtual void SharedItemDelegate::paint(QPainter*, const QStyleOptionViewItem&, const QModelIndex&) const':
/var/tmp/portage/net-misc/seafile-client-7.0.9/work/seafile-client-7.0.9/src/ui/private-share-dialog.cpp:1036:22: error: aggregate 'QPainterPath path' has incomplete type and cannot be defined
 1036 |         QPainterPath path;
      |                      ^~~~
ninja: build stopped: subcommand failed.
 * ERROR: net-misc/seafile-client-7.0.9::testworld failed (compile phase):
 *   ninja -v -j16 -l16 failed
```

I don't do Qt but maybe it's a missing dep?

**Comment by CaseOf** [comment]
@juippis are you building with Qt 5.15? It's known to not work.

**Comment by juippis** [comment]
Yes, I've literally tried by emerging seafile-client. If QT-5.15 doesn't work, you'll have to restrict the version in the ebuild for `<5.15` until it gets fixed.

File: net-misc/seafile/seafile-7.0.9.ebuild [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	default
+	sed -i -e 's/valac /${VALAC} /' lib/Makefile.am || die
+	eautoreconf
+	vala_src_prepare
+}
+
+src_install() {
+	default
+	# Remove unnecessary .la files, as recommended by ltprune.eclass
+	find "${ED}" -name '*.la' -delete || die
```
Comment (jsmolic) [inline net-misc/seafile/seafile-7.0.9.ebuild:45]: Please delete `.a` files in 7.0.9 version as well ([#739710](https://bugs.gentoo.org/739710))

**Comment by CaseOf** [comment]
fixed Qt5 deps for <5.15 and *.a files.

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**